client/apps/landing/src/hooks/use-marketplace-approval.tsx

```
@@ -3,60 +3,125 @@ import { SeasonPassAbi } from "@bibliothecadao/eternum";
 
     setIsApprovingMarketplace(true);
     try {
       const data = await sendAsync();
-      console.log(data);
       if (data?.transaction_hash) {
-        setIsApproved(true);
-        setIsApprovingMarketplace(false);
+        // Mark all collections as approved after successful transaction
+        const newApprovedCollections = collectionAddresses.reduce(
+          (acc, addr) => ({
+            ...acc,
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)